contracts/IntentSource.sol

```diff
@@ -273,45 +276,38 @@ contract IntentSource is IIntentSource, Semver {
 
             emit Withdrawal(intentHash, claimant);
 
-            new Vault{salt: routeHash}(intentHash, reward);
+            new Vault{salt: routeHash}(intentHash, _intent.reward);
 
             return;
         }
 
         if (claimant == address(0)) {
             revert UnauthorizedWithdrawal(intentHash);
+        } else if (destinationChainID != uint96(_intent.route.destination)) {
+            revert WrongDestinationChain(intentHash);
```

Comment:
Should we try to reset the storage in prover and save the right chain id here? Instead of reverting?